src/radius_search_filter.cpp

```diff
@@ -52,6 +52,11 @@ bool laser_filters::RadiusSearchFilter::update(
     }
 
     int counter = 0;
+
+    // adaptive threshold based on input_scan range value
+    // the further the point, the larger the radius search is set
+    double cur_threshold = ((int)(input_scan.ranges[i]) + 1) * threshold_ratio_;
```

Comment:
The +1 is for the reason described below: 
0 - 0.99 m -> threshold would be 1 * radius_ratio_
0.1- 1.99 m -> 2 * radius_ratio_ ...
Also, I have modified the radius filter to render euclidean distance. I added three test cases in the package to verify correctness.